main.ts

```diff
@@ -52,82 +75,79 @@ export default class HotkeysForTemplates extends Plugin {
 
   pushCommand(templateFile: TemplateFile) {
     if (this.getFile(templateFile)) {
-      if (templateFile.plugin === "core") {
-        this.addCommand({
-          id: templateFile.path,
-          name: `Insert: ${templateFile.path.replace(".md", "")}`,
-          callback: () => this.coreInsertTemplate(templateFile)
-        });
-      } else {
-        this.addCommand({
-          id: templateFile.plugin + ":" + templateFile.path,
-          name: `Insert from Templater: ${templateFile.path.replace(".md", "")}`,
-          callback: () => this.templaterInsertTemplate(templateFile)
-        });
+      switch(templateFile.plugin) {
+        case 'core':
+          this.addCommand({
+            id: templateFile.path,
+            name: `Insert: ${templateFile.path.replace(".md", "")}`,
+            callback: () => this.coreInsertTemplate(templateFile)
+          });
+          break;
+        case 'templater-obsidian':
+          this.addCommand({
+            id: templateFile.plugin + ":" + templateFile.path,
+            name: `Insert from Templater: ${templateFile.path.replace(".md", "")}`,
+            callback: () => this.templaterInsertTemplate(templateFile)
+          });
+          break;
+        default:
+          new Notice(this.manifest.name + ': Unknown plugin type for ' + templateFile.path);          
+          return;
       }
     } else {
-      if (templateFile.plugin === "core") {
-        this.settings.files.remove(templateFile.path);
-      } else {
-        this.settings.templaterFiles.remove(templateFile.path);
+      switch(templateFile.plugin) {
+        case 'core':
+          this.settings.coreFiles.remove(templateFile.path);
+          break;
+        case 'templater-obsidian':
+          this.settings.templaterFiles.remove(templateFile.path);
+          break;
+        default:
+          new Notice(this.manifest.name + ': Unknown plugin type for ' + templateFile.path);
+          return;
       }
       this.saveSettings();
     }
   }
 
-
   coreInsertTemplate(fileName: TemplateFile): void {
     const file = this.getFile(fileName);
-
-    if (!(file instanceof TFile)) {
+    let filePath = this.app.vault.getAbstractFileByPath(file as any);
+    if (!(filePath instanceof TFile)) {
       new Notice("Cannot find file");
       return;
+    } else {
+      this.corePlugin.instance.insertTemplate(filePath);
     }
-
-    this.corePlugin.insertTemplate(file);
   }
 
   templaterInsertTemplate(fileName: TemplateFile): void {
     const file = this.getFile(fileName);
-
-    if (!(file instanceof TFile)) {
+    let filePath = this.app.vault.getAbstractFileByPath(file as any);

```

Comment:
`getAbstractFileByPath()` returns `TAbstractFile`which can either be `TFile` or `TFolder`. So the variable is not the path. And why do you use `file as any` since `getFile()` returns a string?